### PR TITLE
Fixes for minor defects with Jenkins flavor of the pipeline

### PR DESCRIPTION
**Pull request by oiavorskyi**
The fixes are provided for the following defects:
* #134 
* #137 
* #138 
* #141

### CODE REVIEW

File: buildSrc/src/test/groovy/org/springframework/cloud/pipelines/ProjectCustomizerSpec.groovy [hunk 1]
```diff
@@ -109,4 +111,8 @@ class ProjectCustomizerSpec extends Specification {
 			}
 	}
 
+	String filePath(String canonicalPath) {
+		canonicalPath.replaceAll("/", Matcher.quoteReplacement(File.separator))
```
Comment (marcingrzejszczak) [inline buildSrc/src/test/groovy/org/springframework/cloud/pipelines/ProjectCustomizerSpec.groovy]: Can you add `return`? I personally always prefer it to be there in Groovy code

File: jenkins/jobs/jenkins_pipeline_sample.groovy [hunk 2]
```diff
@@ -43,11 +44,24 @@ String mySqlRootCredential = binding.variables["MYSQL_ROOT_CREDENTIAL_ID"] ?: ""
 String mySqlCredential = binding.variables["MYSQL_CREDENTIAL_ID"] ?: ""
 // remove::end[K8S]
 
+ScmContext.metaClass.configureScm = { repoId, branchId ->
```
Comment (marcingrzejszczak) [inline jenkins/jobs/jenkins_pipeline_sample.groovy]: Hmm I think we can try to rewrite it in order no to use meta programming. WDYT?

Comment (marcingrzejszczak) [reply jenkins/jobs/jenkins_pipeline_sample.groovy]: Sth like (I don't really remember the correct classes)

```groovy
Closure configureScm = { DslContext context, String repoId, String branchId ->  
context.git {
		remote {
			name('origin')
			url(repoId)
			branch(branchId)
			credentials(gitUseSshKey ? gitSshCredentials : gitCredentials)
		}
		extensions {
			wipeOutWorkspace()
		}
	}
}
```

then you would call it like this

```groovy
configureScm(delegate as DslContext, fullGitRepo, branchName)
```

File: jenkins/jobs/jenkins_pipeline_sample.groovy [hunk 3]
```diff
@@ -399,7 +369,7 @@ parsedRepos.each {
 					allowEmpty()
 					// remove::end[CF]
 				}
-				// end::start[K8S]
+				// remove::end[K8S]
```
Comment (marcingrzejszczak) [inline jenkins/jobs/jenkins_pipeline_sample.groovy]: thanks :)

File: jenkins/jobs/jenkins_pipeline_sample.groovy (hunk 2 above)
Comment (oiavorskyi) [reply jenkins/jobs/jenkins_pipeline_sample.groovy]: We could definitely do it if you have reservations about metaprogramming. The reason I like it in this scenario is that it basically uses the same mechanism as the Jenkins DSL itself thus making it look more natural. It also guarantees that we could only call the closure from the proper DSL block only and the error messages are more natural.

What are your concerns about adding the closure directly to the ScmContext?

File: buildSrc/src/test/groovy/org/springframework/cloud/pipelines/ProjectCustomizerSpec.groovy (hunk 1 above)
Comment (oiavorskyi) [reply buildSrc/src/test/groovy/org/springframework/cloud/pipelines/ProjectCustomizerSpec.groovy]: Absolutely!

File: jenkins/jobs/jenkins_pipeline_sample.groovy [hunk 4]
```diff
@@ -43,11 +44,24 @@ String mySqlRootCredential = binding.variables["MYSQL_ROOT_CREDENTIAL_ID"] ?: ""
 String mySqlCredential = binding.variables["MYSQL_CREDENTIAL_ID"] ?: ""
 // remove::end[K8S]
 
+Closure configureScm = { ScmContext context, String repoId, String branchId ->
```
Comment (marcingrzejszczak) [inline jenkins/jobs/jenkins_pipeline_sample.groovy]: Bootifull!

**Pull request merged**